src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -178,30 +178,31 @@ class MessageBufferProcessor(
 		} yield messages.length
 	}
 
+	private def deleteMessage(msg: MessageLike)(implicit ec: ExecutionContext): Future[Unit] = msg match {
+		case FailedMessage(id, _, _, _, _) =>
+			messageStore.deleteFailedMessage(
+				id.getOrElse(throw new IllegalStateException("Got a message without an id from database"))
+			)
+		case Message(_, _, _, channelId, deliveryTag, _) =>
+			messageStore.deleteMessage(channelId, deliveryTag).map(_ => ())
+	}
+
 	/**
 	 * Resend the messages in the list and if managed to publish, deletes the message
 	 */
 	private def resendAndDelete(
-		msgs: List[Message],
+		msgs: List[MessageLike],
```

Comment:
Can msgs be anything else than `List[FailedMessage]` in this case?

Reply:
I don't see why not. The message was sent successfully, but the confirmation never came, so, we are resending it.